apps/web/composables/useModernImage/__tests__/useModernImage.spec.ts

```diff
@@ -0,0 +1,111 @@
+        expect(res).toBe('https://example.com/item/images/image.jpg.webp');
+    });
+
+    it('should not add the webp extension to the url if the url has no /item/images/ in it', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/images/image.jpg');
+    });
+
+    it('should not add the webp extension to the url if the base extension is already webp', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.webp';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.webp');
+    });
+});
+
+describe('useModernImage with webp and avif enabled', () => {
+    beforeEach(() => {
+        useRuntimeConfig.mockImplementation(() => {
+            return {
+                public: {
+                    useAvif: true,
+                    useWebp: true
+                }
+            }
+        });
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+    });
+
+    it('should add the avif extension to the url', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.jpg.avif');
+    });
+
+    it('should not add the avif extension to the url if the base extension is already avif', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.avif';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.avif');
+    });
+
+    it('should not add the avif extension to the url if the url has no /item/images/ in it', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/images/image.jpg');
+    });
+    
+    it('should not add the avif extension to the url if the base extension is not supported', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.svg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.svg');
+    });
+
+    it('should add the avif extension to the url if both avif and webp are enabled', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.jpg.avif');
+    });
+
+});
```

Comment:
**question**: Should we add a third suite of tests for a case where the browser supports neither AVIF nor WebP?

Reply:
done, its not about the browser support, we just check the settings.